File: services/client/src/protocol/message.go
package protocol

type MessageType uint8

const (
	MsgStartTransmission MessageType = 0x01
	MsgBet               MessageType = 0x02
	MsgEndTransmission   MessageType = 0x03
	MsgOk                MessageType = 0x10
	MsgWinners           MessageType = 0x11
	MsgError             MessageType = 0x12
)

const (
	_TYPE_FIELD_SIZE   = 1
	_LENGTH_FIELD_SIZE = 2
	_HEADER_SIZE       = _TYPE_FIELD_SIZE + _LENGTH_FIELD_SIZE
	_MAX_UINT_16       = 65535
	_MAX_PAYLOAD_SIZE  = _MAX_UINT_16
)

type Message struct {
	Type    MessageType
	Payload []byte
}

func NewMessage(messageType MessageType, payload []byte) *Message {
	return &Message{
		messageType,
		payload,
	}
}

func (messageType MessageType) String() string {
	switch messageType {
	case MsgStartTransmission:
		return "START_TRANSMISSION"
	case MsgBet:
		return "BET"
	case MsgEndTransmission:
		return "END_TRANSMISSION"
	case MsgOk:
		return "OK"
	case MsgWinners:
		return "WINNERS"
	case MsgError:
		return "ERROR"
	default:
		return "UNKNOWN"
	}
}

File: services/server/src/protocol/message.py
_TYPE_FIELD_SIZE = 1
_LENGTH_FIELD_SIZE = 2
_HEADER_SIZE = _TYPE_FIELD_SIZE + _LENGTH_FIELD_SIZE

_MAX_UINT_16 = 65535
_MAX_PAYLOAD_SIZE = _MAX_UINT_16

_UNKNOWN_TYPE_NAME = "UNKNOWN"


class MessageType:
    START_TRANSMISSION = 0x01
    BETS = 0x02
    END_TRANSMISSION = 0x03
    OK = 0x10
    WINNERS = 0x11
    ERROR = 0x12

    @classmethod
    def get_name_from_code(cls, code: int) -> str:
        return _TYPE_NAMES.get(code, _UNKNOWN_TYPE_NAME)


_TYPE_NAMES = {
    MessageType.START_TRANSMISSION: "START_TRANSMISSION",
    MessageType.BETS: "BETS",
    MessageType.END_TRANSMISSION: "END_TRANSMISSION",
    MessageType.OK: "OK",
    MessageType.WINNERS: "WINNERS",
    MessageType.ERROR: "ERROR",
}


class Message:
    def __init__(self, message_type: int, payload: bytes = b"") -> None:
        self.type = message_type
        self.payload = payload
